ingest/ingest_pipeline.py

```diff
@@ -576,24 +576,29 @@ def main() -> None:
     Returns:
         None
     """
-    parsed_args = create_parser().parse_args()
-    validate_arguments(parsed_args)
-    arguments = vars(parsed_args)
-    # Initialize global variables for current ingest job
-    config.init(
-        arguments["study_id"],
-        arguments["study_file_id"],
-        arguments["user_metrics_uuid"],
-    )
-    ingest = IngestPipeline(**arguments)
-    status, status_cell_metadata = run_ingest(ingest, arguments, parsed_args)
-    # Print metrics properties
-    metrics_dump = config.get_metric_properties().get_properties()
-    for key in metrics_dump.keys():
-        print(f'{key}: {metrics_dump[key]}')
-
-    # Log Mixpanel events
-    MetricsService.log(config.get_parent_event_name(), config.get_metric_properties())
+    status = 0
+    try:
+        parsed_args = create_parser().parse_args()
+        validate_arguments(parsed_args)
+        arguments = vars(parsed_args)
+        # Initialize global variables for current ingest job
+        config.init(
+            arguments["study_id"],
+            arguments["study_file_id"],
+            arguments["user_metrics_uuid"],
+        )
+        ingest = IngestPipeline(**arguments)
+        status, status_cell_metadata = run_ingest(ingest, arguments, parsed_args)
+        # Print metrics properties
+        metrics_dump = config.get_metric_properties().get_properties()
+        for key in metrics_dump.keys():
+            print(f'{key}: {metrics_dump[key]}')
+
+        # Log Mixpanel events
+        MetricsService.log(config.get_parent_event_name(), config.get_metric_properties())
+    except Exception as e:
+        log_exception(IngestPipeline.dev_logger, IngestPipeline.user_logger, e)
```

Comment:
Am I right that this logs only to files, and not to Sentry nor Mixpanel?

Reply:
this logs to Sentry and to files, but not to Mixpanel